### ISSUE TITLE
SSE provider not disconnected when window loses focus

### ISSUE THREAD

**Issue by alxgrk** [DevCycleHQ/js-sdks#469]
Hi,

more or less related to https://github.com/DevCycleHQ/js-sdks/issues/468, when playing around with js-sdk for Angular, I noticed that sometimes in contrast to what is described [here](https://docs.devcycle.com/sdk/features/realtime-updates#javascript-sdk-react-sdk--react-native-sdk), the SSE connection is not dropped when the user de-focuses the current page and a new one is created when returning to the page resulting in two open connections.

On other hand, in another situation the connection was open all the time regardless of whether the page is focused or not.

Are you able to somehow reproduce this on your side? Or put differently, what would be the expected behavior?

Best regards,
Alex

**Comment by cdruxerman** [comment]
Hi Alex, our team is looking into this, and we'll keep you posted.

**Comment by jsalaber** [comment]
Hi @alxgrk, seems like our docs could be a little more clearer on this. As it stands, the wording implies that losing focus will immediately disconnect the streaming connection and will reconnect once the page is back in view. This is true, but only after the some delay has passed, the default of which is set to 2 minutes. This is to prevent connecting and reconnecting if the user switches between tabs quickly. 

This can be configured in the start options using the option `inactivityDelay`. You can also set the default log level to `debug` to see what kind of logs you can find; the log `'Page is not visible, closing streaming connection'` will occur if the user is not focused on the page for more than `inactivityDelay`, and `'Page became visible, refetching config'` will occur once they refocus on the page.

If you also have a test project you can reproduce this with that would be helpful, I myself have tested this behaviour on Angular's [https://angular.io/tutorial/tour-of-heroes/toh-pt0]("Tour of Heroes") project and it seems to work as expected.

**Comment by jsalaber** [comment]
My mistake @alxgrk, while we do have that being passed down from the config, I believe we made a conscious decision to not allow it to be configured as it impacts our realtime messaging provider usage and could potentially open us up to huge costs. We might revisit this at a later date.

I've updated the documentation to be clearer on the delay before closing the connection, so I'll close this ticket out. Thanks!
